File: src/main/java/ru/ioleksiv/telegram/bot/core/api/model/method/passport/SetPassportDataErrors.java
package ru.ioleksiv.telegram.bot.core.api.model.method.passport;

import com.fasterxml.jackson.annotation.JsonProperty;
import ru.ioleksiv.telegram.bot.core.api.model.objects.passport.errors.PassportElementError;
import ru.ioleksiv.telegram.bot.core.api.model.NetworkError;
import ru.ioleksiv.telegram.bot.core.model.method.RunnableAction;
import ru.ioleksiv.telegram.bot.core.model.responses.CommonResponse;
import ru.ioleksiv.telegram.bot.core.model.responses.ResponseCollection;
import ru.ioleksiv.telegram.bot.core.controller.network.Networker;

import java.util.List;

/**
 * @see <a href="https://core.telegram.org/bots/api#setpassportdataerrors>setPassportDataErrors</a>
 */
public class SetPassportDataErrors extends RunnableAction<Boolean> {
    private static final String METHOD = "setPassportDataErrors";
    /**
     * user_id	Integer	Yes	User identifier
     */
    @JsonProperty("user_id")
    private Integer userId = null;
    /**
     * errors	Array of PassportElementError	Yes	A JSON-serialized array describing the errors
     */
    @JsonProperty("errors")
    private List<PassportElementError> errors = null;

    public SetPassportDataErrors(Networker networker) {
        super(METHOD, networker);
    }

    public Integer getUserId() {
        return userId;
    }

    public void setUserId(Integer userId) {
        this.userId = userId;
    }

    @Override
    protected Class<? extends CommonResponse<Boolean>> getResultWrapperClass() {
        return ResponseCollection.BooleanResponse.class;
    }

    public List<PassportElementError> getErrors() {
        return errors;
    }

    public void setErrors(List<PassportElementError> errors) {
        this.errors = errors;
    }

    @Override
    public SetPassportDataErrors setNetworkErrorListener(NetworkError onNetworkError) {
        pSetNetworkErrorListener(onNetworkError);
        return this;
    }
}

File: server-api/src/main/kotlin/io/github/oleksivio/tl/kbot/server/api/method/passport/SetPassportDataErrors.kt
package io.github.oleksivio.tl.kbot.server.api.method.passport

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.tl.kbot.server.api.model.ApiDict
import io.github.oleksivio.tl.kbot.server.api.model.BooleanResponse
import io.github.oleksivio.tl.kbot.server.api.model.method.Action
import io.github.oleksivio.tl.kbot.server.api.objects.passport.errors.PassportElementError

/**
 * @see [setPassportDataErrors](https://core.telegram.org/bots/api/#setpassportdataerrors)
 */
data class SetPassportDataErrors(
    /**
     * user_id Integer Yes User identifier
     */
    @JsonProperty(ApiDict.USER_ID_KEY)
    val userId: Int,
    /**
     * errors Array of PassportElementError Yes A JSON-serialized array describing the errors
     */
    @JsonProperty(ApiDict.ERRORS_KEY)
    val errors: List<PassportElementError>

) : Action<Boolean>() {

    @JsonProperty(ApiDict.METHOD_KEY)
    override val method = "setPassportDataErrors"
    @JsonIgnore
    override val resultWrapperClass = BooleanResponse::class
}
